pkg/controller/jobs/job/job_controller.go

```diff
@@ -216,6 +216,24 @@ func (j *Job) PodSets() []kueue.PodSet {
 	}
 }
 
+// For prebuilt workloads we should preserve the following template labels since they
+// are generated by the batch/job controller upon job object creation, and only used
+// by it to identify the jobs pods.
+var (
+	labelPrefix         = "batch.kubernetes.io/"
+	jobNameSuffix       = "job-name"
+	controllerUidSuffix = "controller-uid"
+	preservedLabelKeys  = []string{jobNameSuffix, controllerUidSuffix, labelPrefix + jobNameSuffix, labelPrefix + controllerUidSuffix}
+)
+
+func addPreservedLabelsToInfo(labels map[string]string, info *podset.PodSetInfo) {
```

Comment:
I see *what* we do - copy a hardcoded subset of labels that to the workload on `RunWithPodSetsInfo`, and copied back to Job on  `RunWithPodSetsInfo`.  

However, I still don't understand *why*. Also, I don't see this code being protected by the presence of prebuilt-workload, so it makes me uncertain if the currently existing functionality is not affected.

Some clarifying questions to help me understand:
- is this code needed for the management cluster Job, or the worker Job (or both?)
- just a suspicion: does the code aim that the UID of the Job on the worker is the same as the UID of the job in management cluster?
- can the scope be narrowed down to only jobs with prebuilt-workload?
- what would happen if we don't do it?
- why we do this for this specific set of labels, and not others?

Reply:
I think we discussed offline at least part of this.

>  is this code needed for the management cluster Job, or the worker Job (or both?)

Not relevant , trying to change them will fail, they are generated and "guarded" by the main batch/job controller/webhook.

> just a suspicion: does the code aim that the UID of the Job on the worker is the same as the UID of the job in management cluster?

No. The UIDs will most likely be different, which will lead to the inability to start the job on the worker.

> can the scope be narrowed down to only jobs with prebuilt-workload?

Maybe but it will not make a difference,  if the workload is not prebuilt, the values are already the same.

>  what would happen if we don't do it?

Trying to set them to something else than the values enforced by the main batch/job controller / webhook  will result in a web-hook error at first.

> why we do this for this specific set of labels, and not others?

These are the labels generated by  main batch/job controller / webhook and used to track the child pods in execution.

Reply:
Sorry, I still don't understand, maybe another offline session will help :).

In the meanwhile, you say:
> Not relevant , trying to change them will fail, they are generated and "guarded" by the main batch/job controller/webhook.

what would be trying to change the values? Can you point to the code, I'm really confused about this part.

> No. The UIDs will most likely be different, which will lead to the inability to start the job on the worker.

This is also confusing to me. Could you please provide maybe snippets for the mainJob and the mirrorJob, containing the example values of the labels, to illustrate what would be the problem / conflict without this change, and how the values of the labels will relate after this change.

Reply:
I see the code is changed in https://github.com/kubernetes-sigs/kueue/pull/1358/commits/697d83efe6d97ca12d9beb49864a6c080b5a7de8. 
The new version is much easier to comprehend. Thank you!